### PR TITLE
Add combinations

### PR DESCRIPTION
**Pull request by obelisk0114**
Return all possible combinations of k numbers out of 1 ... n.

### CODE REVIEW

File: backtracking/all_combinations.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+def print_all_state(total_list):
+    for i in total_list:
+        print(*i)
+
+
+n = 4
+k = 2
+total_list = generate_all_combinations(n, k)
+print_all_state(total_list)
```
Comment (cclauss) [inline backtracking/all_combinations.py]: Please put these four lines under a `if __name__ == '__main__'`:
```python
if __name__ == '__main__':
    n = 4
    k = 2
    total_list = generate_all_combinations(n, k)
    print_all_state(total_list)
```

This should not change the functionality of your script but it should give it a new superpower:
* __python3 -m doctest -v backtracking/all_combinations.py__

**Comment by cclauss** [comment]
It would be cool to think up more helpful variable names than __n__ and __k__.

**Comment by cclauss** [comment]
Is this ready to be merged or are you still working on it?

**Comment by cclauss** [comment]
Tests fail on https://travis-ci.org/TheAlgorithms/Python/builds/558436341#L649

**Comment by cclauss** [comment]
Is this ready to be merged or are you still working on it?

**Comment by obelisk0114** [comment]
It's ready to be merged.

**Comment by cclauss** [comment]
import itertools ; print(list(itertools.combinations(range(1, 5), 2)))

**Pull request merged**